magpy/gui/dialogclasses.py

```diff
@@ -2286,6 +2286,147 @@ def OnLoadDIRemote(self,e):
+        boxSizer = wx.BoxSizer(orient=wx.HORIZONTAL)
+
+        # Prepare some reusable arguments for calling sizer.Add():
+        expandOption = dict(flag=wx.EXPAND)
+        noOptions = dict()
+        emptySpace = ((0, 0), noOptions)
+
+        # Add the controls to the sizers:
+        controls = [(self.dateInfoText, noOptions),
+                (self.startText, noOptions),
+                (self.startDatePicker, dict(flag=wx.ALIGN_CENTER)),
+                (self.endText, noOptions),
+                (self.endDatePicker, dict(flag=wx.ALIGN_CENTER)),
+                emptySpace,
+                emptySpace,
+                (self.extInfoText, noOptions),
+                (self.extensionText, noOptions),
+                (self.extensionComboBox,  dict(flag=wx.ALIGN_CENTER)),
+                (self.analyzeCheckBox, dict(flag=wx.ALIGN_CENTER)),
+                emptySpace,
+                emptySpace,
+                (self.okButton, dict(flag=wx.ALIGN_CENTER)),
+                (self.closeButton, dict(flag=wx.ALIGN_CENTER))]
+
+        # A GridSizer will contain the other controls:
+        cols = 1
+        rows = int(np.ceil(len(controls)/float(cols)))
+        gridSizer = wx.FlexGridSizer(rows=rows, cols=cols, vgap=10, hgap=10)
+
+        for control, options in controls:
+            gridSizer.Add(control, **options)
+
+        for control, options in \
+                [(gridSizer, dict(border=5, flag=wx.ALL))]:
+            boxSizer.Add(control, **options)
+
+        self.SetSizerAndFit(boxSizer)
+
+    def bindControls(self):
+        self.extensionComboBox.Bind(wx.EVT_CHOICE, self.onExtend)
+        self.startDatePicker.Bind(wx.EVT_DATE_CHANGED, self.onChangeDate)
+        self.endDatePicker.Bind(wx.EVT_DATE_CHANGED, self.onChangeDate)
+
+    def onChangeDate(self, e):
+        self.createUrls()
+
+    def onExtend(self, e):
+        startvalue = self.startDatePicker.GetValue()
+        starttime = datetime.fromtimestamp(startvalue.GetTicks())
+        endvalue = self.endDatePicker.GetValue()
+        endtime = datetime.fromtimestamp(endvalue.GetTicks())
+        extension = self.extensionComboBox.GetString(self.extensionComboBox.GetSelection())
+        dt = extension[0]
+        dttype = extension[2]
+        if dttype == 'd':
+            dt = int(dt)
+        if dttype == 'w':
+            dt = int(dt) * 7
+        starttime = starttime - timedelta(days = dt)
+        endtime = endtime + timedelta(days = dt)
+        start = wx.DateTimeFromTimeT(time.mktime(starttime.timetuple()))
+        end = wx.DateTimeFromTimeT(time.mktime(endtime.timetuple()))
+        self.startDatePicker.SetValue(start)
+        self.endDatePicker.SetValue(end)
+        self.createUrls()
+
+    def createUrls(self):
+        self.urls = []
+        obsid = self.stream.header.get('StationID', '')
+        extension = self.extensionComboBox.GetString(self.extensionComboBox.GetSelection())
+        dt = extension[0]
```

Comment:
Is extension used in this method?

Does the list of urls need to be stored, or could this method be getUrls() and called after the dialog is closed?